### DISCUSSION TITLE
Is it possible to parse `#[my_attribute("literal")]`?

### DISCUSSION THREAD

**Discussion by gavrilikhin-d** [TedDriggs/darling#240] [Q&A] [answered]
How can I create derive macro `Rule` that is able to parse `#[rule("{username}:{password}")]` as its attribute?
```rust
use derive::Rule;

#[derive(Rule)]
#[rule("{username}:{password}")]
struct Credentials {
       #[pattern("/[_0-9a-zA-Z]+/")]
       pub username: String,
       #[pattern("/[_0-9a-zA-Z]+/")]
       pub password: String
}
```

**Comment by TedDriggs** [comment, marked as answer]
You’ll want to follow the examples around attribute macros that aren’t derive macros, and probably use the `forward_attrs` option to collect those for subsequent processing.
